pkg/docker/docker.go

```diff
@@ -931,27 +906,23 @@ func (d *stiDocker) RunContainer(opts RunContainerOptions) error {
 
 	// Create a new container.
 	glog.V(2).Infof("Creating container with options {Name:%q Config:%+v HostConfig:%+v} ...", createOpts.Name, createOpts.Config, createOpts.HostConfig)
-	var container *dockertypes.ContainerCreateResponse
-	if err = util.TimeoutAfter(DefaultDockerTimeout, "timeout after waiting %v for Docker to create container", func() error {
-		var createErr error
-		if createOpts.HostConfig != nil && createOpts.HostConfig.ShmSize <= 0 {
-			createOpts.HostConfig.ShmSize = DefaultShmSize
-		}
-		container, createErr = d.kubeDockerClient.CreateContainer(createOpts)
-		return createErr
-	}); err != nil {
+	ctx, cancel := getDefaultContext(DefaultDockerTimeout)
+	defer cancel()
+	if createOpts.HostConfig != nil && createOpts.HostConfig.ShmSize <= 0 {
+		createOpts.HostConfig.ShmSize = defaultShmSize
+	}
+	container, err := d.client.ContainerCreate(ctx, createOpts.Config, createOpts.HostConfig, createOpts.NetworkingConfig, createOpts.Name)
+	if err != nil {
 		return err
 	}
 
-	containerName := containerNameOrID(container)
-
 	// Container was created, so we defer its removal, and also remove it if we get a SIGINT/SIGTERM/SIGQUIT/SIGHUP.
 	removeContainer := func() {
-		glog.V(4).Infof("Removing container %q ...", containerName)
-		if err := d.RemoveContainer(container.ID); err != nil {
-			glog.V(0).Infof("warning: Failed to remove container %q: %v", containerName, err)
+		glog.V(4).Infof("Removing container %q ...", container.ID)
+		if removeErr := d.RemoveContainer(container.ID); removeErr != nil {
```

Comment:
i used to get beat up a lot when first joining the team about having differen variables for different errors vs. reusing `err`, with basically it didn't golang stylistic conventions

i won't out and out call for a change, but be prepared others might be more adamant about this